crates/goose/src/agents/agent.rs

```diff
@@ -220,6 +234,21 @@ impl Agent {
         }
     }
 
+    /// Get the provider name
+    pub fn provider_name(&self) -> String {
+        // Get the provider name from config
+        let config = Config::global();
+        config
+            .get_param::<String>("GOOSE_PROVIDER")
+            .unwrap_or_else(|_| "unknown".to_string())
+    }
+
+    /// Enable middle-out transform for OpenRouter provider
+    pub fn enable_middle_out_transform(&self) {
```

Comment:
I don't think this should be in the agent, is too specific to openrouter at the moment